ui/src/app/views/sign-in-flow-template/component.tsx

```diff
@@ -0,0 +1,126 @@
+}
+
+interface PageTemplateState {
+  currentStep: string;
+  invitationKey: string;
+}
+
+const pageImages = {
+  'login': {
+    backgroundImgSrc: '/assets/images/login-group.png',
+    smallerBackgroundImgSrc: '/assets/images/login-standing.png'
+  },
+  'invitationKey': {
+    backgroundImgSrc: '/assets/images/invitation-female.png',
+    smallerBackgroundImgSrc: '/assets/images/invitation-female-standing.png'
+  },
+  'createAccount': {
+    backgroundImgSrc: '/assets/images/create-account-male.png',
+    smallerBackgroundImgSrc: '/assets/images/create-account-male-standing.png'
+  }};
+
+const headerImg = '/assets/images/logo-registration-non-signed-in.svg';
+
+const SignPageTemplateReact = withWindowSize()(
+  class extends React.Component<PageTemplateProps, PageTemplateState> {
+
+    constructor(props: PageTemplateProps) {
+      super(props);
+      this.state = {
+        currentStep: 'login',
+        invitationKey: ''
+      };
+      this.setCurrentStep = this.setCurrentStep.bind(this);
+      this.onKeyVerified = this.onKeyVerified.bind(this);
+    }
+
+    nextDirective(index) {
+      switch (index) {
+        case 'login':
+          return <LoginReactComponent signIn={this.props.signIn} onCreateAccount={() =>
+                                     this.setCurrentStep('invitationKey')}/>;
+        case 'invitationKey':
+          return <InvitationKeyReact onInvitationKeyVerify={(key) => this.onKeyVerified(key)}/>;
+        // case 'accountCreation': return <AccountCreationReact setCurrentStep={this.setCurrentStep}
+        default:
+          return;
+        }
+      }
+
+    setCurrentStep(nextStep) {
+      this.setState({
+        currentStep: nextStep
+      });
+    }
+
+    onKeyVerified(invitationKey) {
+      this.setState({
+        invitationKey: invitationKey,
+        currentStep: 'createAccount'
+      });
+    }
+
+    render() {
+      return <div style={styles.signedInContainer}>
+        <div style={{width: '100%', display: 'flex', flexDirection: 'column'}}>
+          <div style={styles.template(this.props.windowSize, pageImages[this.state.currentStep])}>
+            <img style={{height: '1.75rem', marginLeft: '1rem', marginTop: '1rem'}}
+                 src={headerImg}/>
+            <div style={{flex: '0 0 41.66667%', maxWidth: '41.66667%', minWidth: '25rem'}}>
+              {this.nextDirective(this.state.currentStep)}
+            </div>
+          </div>
+        </div>
+      </div>;
+    }
+});
+
+export default SignPageTemplateReact;
+
+@Component({
+  template: '<div #root></div>'
+})
+export class SignInTemplateComponent extends ReactWrapperBase implements OnInit {
+
+  constructor(private signInService: SignInService, private router: Router) {
+    super(SignPageTemplateReact, ['signIn']);
+    this.signIn = this.signIn.bind(this);
+  }
+
+  ngOnInit(): void  {
+    document.body.style.backgroundColor = '#e2e3e5';
+
+    this.signInService.isSignedIn$.subscribe((signedIn) => {
```

Comment:
Since these wrappers are temporary, we should avoid putting logic in them if at all possible. This subscription logic could go into lifecycle methods in the React component. Subscribe in `componentDidMount` and unsubscribe in `componentWillUnmount`.